mood-analysis.rb

```diff
@@ -19,6 +21,39 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(word)
+  word.gsub!(/[!.,#]/, "") unless word.nil?
+end
+
+
+def happy_days(text)
+  days = 0
+  index = analyze_text(text).each_with_index do |mood, i|
+    days += 1 if mood.include? ":-)"
+    break i if days == 3
+  end
+
+  return "There aren't 3 happy days!" if days < 3
+  "It takes #{index} entries for 3 happy days to occur."
+end
+
+def analyze_text(text)
+  text.map do |entry|
+    "#{entry[0,5]} " + analyze_mood(entry)
+  end
+end
+
+def overall_mood(text)
+  mood_count = Hash.new { |hash, key| hash[key] = 0 }
+
+  analyze_text(text).each do |mood|
+    mood_count[mood[-3, 3]] += 1
+  end
+
+  highest_mood = mood_count.max_by { |k, v| v }
+  "The most common mood is #{highest_mood[0]}"
+end
```

Comment:
Here also, I'd rather this just return ":-)" or whatever, but not the entire string answer. Gives us more flexibility in doing something different with this method's result.